spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -113,6 +113,16 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("Integral Division Overflow Handling Matches Spark Behavior") {
+    withTable("t1") {
+      val value = Long.MinValue
+      sql("create table t1(c1 long, c2 short) using parquet")
+      sql(s"insert into t1 values($value, -1)")
+      val res = sql("select c1 div c2 from t1 order by c1")
+      checkSparkAnswer(res)
```

Comment:
We need to assert that the plan is actually running natively in Comet and not falling back to Spark.

```suggestion
      checkSparkAnswerAndOperator(res)
```

Reply:
The cast operation to decimal type is falling back to Spark which led me to make this change given we dont have a comet native cast implementation . Do you think it is a good idea to implement cast operation in comet and undo the above changes ?